dali/operators/audio/nonsilence_op_gpu.cu

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/common/find/find_first_last_gpu.cuh"
+#include "dali/kernels/reduce/reduce_gpu.h"
+#include "dali/kernels/signal/decibel/decibel_calculator.h"
+#include "dali/kernels/signal/moving_mean_square_gpu.h"
+#include "dali/operators/audio/nonsilence_op.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+namespace {
+
+struct threshold_cutoff_db {
+  float thresh_;
```

Comment:
This predicate mixes two things:
- the predicate (which is basicaly `threshold`) - such predicate could find use elsewhere
- the calculation of the threshold value (which for the predicate itself is irrelevant)

The fusion of these functions makes it much less reusable.

Reply:
The points was not for this particular struct to be reusable but rather to be convenient to use in this operator. Would you prefer if the calculation of the threshold was done on the caller side? (I have no problem with that)